src/derivative_operators/multi_dim_bc_operators.jl

```diff
@@ -0,0 +1,161 @@
+
+abstract type MultiDimensionalBC{T, N} <: AbstractBC{T} end
+
+
+@noinline function _slice_rmul!(u_temp::AbstractArray{T,N}, A::AbstractDiffEqLinearOperator, u::AbstractArray{T,N}, dim::Int, pre, post) where {T,N}
+    for J in post
+        for I in pre
+            u_temp[I, :, J] = A*u[I, :, J]
+        end
+    end
+    u_temp
+end
+
+function slice_rmul(A::AbstractDiffEqLinearOperator, u::AbstractArray{T,N}, dim::Int) where {T,N}
+    @assert N != 1
+    u_temp = similar(u)
+
+    _slice_rmul!(u_temp, A, u, dim, CartesianIndices(axes(u)[1:dim-1]), CartesianIndices(axes(u)[(dim+1):end]))
+
+    return u_temp
+end
+
+@noinline function _slice_rmul!(lower::AbstractArray, upper::AbstractArray, A::AbstractArray{B,M}, u::AbstractArray{T,N}, dim::Int, pre, post) where {T,B,N,M}
+    for J in post
+        for I in pre
+
+            tmp = A[I,J]*u[I, :, J]
+            lower[I,J], upper[I,J] = tmp.l, tmp.r
+        end
+    end
+    return (lower, upper)
+end
+
+function slice_rmul(A::AbstractArray{B,M}, u::AbstractArray{T,N}, dim::Int) where {T, B, N,M}
+    @assert N != 1
+    @assert M == N-1
+    lower = zeros(T,perpsize(u,dim))
+    upper = zeros(T,perpsize(u,dim))
+
+    _slice_rmul!(lower, upper, A, u, dim, CartesianIndices(axes(u)[1:dim-1]), CartesianIndices(axes(u)[(dim+1):end]))
+
+    return (lower, upper)
+end
+
+"""
+slicemul is the only limitation on the BCs here being used up to arbitrary dimension, an N dimensional implementation is needed.
+"""
+
```

Comment:
This docstring won't work with this space.

Reply:
That docstring is old and should be gone - I'll add that to the list

Reply:
this one is still here 😛

Reply:
lol